src/lib/sections/Navigation/Logo/Logo.tsx

```diff
@@ -1,15 +1,23 @@
-import { ReactNode } from "react";
+import { ComponentProps, ElementType, ReactNode } from "react";
 
-export interface NavigationLogoProps {
+import classNames from "classnames";
+
+import { AsProp } from "@/lib/sections/Navigation/types";
+
+export interface NavigationLogoProps extends ComponentProps<typeof Logo> {
   children: ReactNode;
 }
 
-export const Logo = ({ children }: NavigationLogoProps) => {
+export const Logo = <
+C extends ElementType = "a",
+T extends ComponentProps<C> = ComponentProps<C>,
+>({ as, children, className, ...props }: AsProp<C> & Omit<T, "as">) => {
+  const Component = as || "a";
    return (
-    <a className="p-panel__logo" href="/">
+    <Component className={classNames("p-panel__logo", className)} {...props}>
```

Comment:
One suggestion I have is we could create a HOC e.g. `withAsProp` to reduce duplication (and reuse in Link component).

Reply:
Did some investigating around this and I think it needs some more time/work. I'll merge this in its current state and create a jira ticket.